BEIMA.Backend/AuthService/AuthenticationService.cs

```diff
@@ -0,0 +1,88 @@
+﻿using BEIMA.Backend.Models;
+using BEIMA.Backend.MongoService;
+using JWT;
+using JWT.Algorithms;
+using JWT.Builder;
+using JWT.Serializers;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BEIMA.Backend.AuthService
+{
+    public sealed class AuthenticationService : IAuthenticationService
+    {
+        private static readonly Lazy<AuthenticationService> instance = new(() => new AuthenticationService());
+
+        private readonly IJwtAlgorithm _algorithm;
+        private readonly IJsonSerializer _serializer;
+        private readonly IBase64UrlEncoder _base64Encoder;
+        private readonly IJwtEncoder _jwtEncoder;
+        
+        private AuthenticationService()
+        {
+            _algorithm = new HMACSHA256Algorithm();
+            _serializer = new JsonNetSerializer();
+            _base64Encoder = new JwtBase64UrlEncoder();
+            _jwtEncoder = new JwtEncoder(_algorithm, _serializer, _base64Encoder);
+        }
+
+        public static AuthenticationService Instance { get { return instance.Value; } }
+
+        public string CreateToken(User user)
+        {
+            var claims = new Claims()
+            {
+                Username = user.Username,
+                Role = user.Role,
+            };
+            var secretKey = Environment.GetEnvironmentVariable("JwtKey");
```

Comment:
So, we will be doing this differently when we get logins working right? Doesn't feel super secure to be holding our secret key in an environment variable.

Reply:
I think this will have to be a bigger discussion as we will want to move the db/storage connection strings out as well. there is a secrets manger that we can use to store theses variables. I'm not sure how it works with github build/deploy though